OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -815,7 +818,39 @@ void UpdatePlayerList()
 			while (players.Children.Count > idx)
 				players.RemoveChild(players.Children[idx]);
 
-			tabCompletion.Names = orderManager.LobbyInfo.Clients.Select(c => c.Name).Distinct().ToList();
+			tabCompletion.Names = lobbyClients.Select(c => c.Name).Distinct().ToList();
+
+			teamArrangementInfo = GetTeamArrangementInfo(lobbyClients.Where(c => c.Slot != null).ToList());
+		}
+
+		string GetTeamArrangementInfo(List<Session.Client> lobbyPlayers)
+		{
+			var teamsInfo = "";
+			var noTeamPlayersCount = lobbyPlayers.Count(c => c.Team == 0);
+			var teams = lobbyPlayers.Where(c => c.Team != 0).GroupBy(c => c.Team).OrderByDescending(g => g.Count()).ToList();
+
+			if (noTeamPlayersCount + teams.Count() > 1 && teams.Any(t => t.Count() > 1))
+			{
+				foreach (var team in teams.Where(t => t.Count() > 1))
+				{
+					if (team != teams.First())
```

Comment:
If this were a `for` loop you wouldn't need to lookup `First()` on every iteration.